test/widgets/action_sheet_test.dart

```diff
@@ -536,4 +536,20 @@ void main() {
       check(mockSharePlus.sharedString).isNull();
     });
   });
+
+  group('MarkAsUnread', () {
+    testWidgets('not visible if message is not read', (WidgetTester tester) async {
+      final unreadMessage = eg.streamMessage(flags: []);
+      await setupToMessageActionSheet(tester, message: unreadMessage, narrow: TopicNarrow.ofMessage(unreadMessage));
+
+      check(find.byIcon(Icons.mark_chat_unread_outlined).evaluate().length).equals(0);
+    });
+
+    testWidgets('visible if message is read', (WidgetTester tester) async {
+      final message = eg.streamMessage(flags: [MessageFlag.read]);
+      await setupToMessageActionSheet(tester, message: message, narrow: TopicNarrow.ofMessage(message));
+
+      check(find.byIcon(Icons.mark_chat_unread_outlined).evaluate().length).equals(1);
```

Comment:
nit:
```suggestion
      check(find.byIcon(Icons.mark_chat_unread_outlined).evaluate()).single;
```

Similarly, the test above can use `isEmpty()`.